### ISSUE TITLE
target_compile_options requires CMake 2.8.12+

### ISSUE THREAD

**Issue by x64architecture** [andlabs/libui#148]
CMake function target_compile_options requires CMake 2.8.12+ and libui targets 2.8.11 so it should be worked around or the minimum cmake version should be incremented.

**Comment by andlabs** [comment]
```
[09:49:44]  <+ngladitz>	then you probably hit https://cmake.org/gitweb?p=cmake.git;a=commit;h=95cdf132489c79e88a10fdf7a7566fa002c7680b
[09:50:16]  <+ngladitz>	the fix in that commit is in cmake since v3.1.0
```
I'll probably just bump the minimum required cmake version to that.

**Comment by ghaberek** [comment]
I am still using Ubuntu 14.04, and I found this AskUbuntu answer helpful to get from CMake 2.8 to 3.2.

http://askubuntu.com/a/610352
